app/components-react/shared/inputs/FormFactory.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect } from 'react';
+import { FormProps } from 'antd/lib/form';
+import debounce from 'lodash/debounce';
+import * as inputs from './inputList';
+import { TSlobsInputProps } from './inputs';
+import Form, { useForm } from './Form';
+import { TInputMetadata } from './metadata';
+
+type TInputValue = string | number | boolean;
+
+const componentTable: Dictionary<React.FunctionComponent<TSlobsInputProps<{}, TInputValue>>> = {
+  text: inputs.TextInput,
+  number: inputs.NumberInput,
+  slider: inputs.SliderInput,
+  bool: inputs.CheckboxInput,
+  list: inputs.ListInput,
+  seconds: inputs.SliderInput,
+  autocomplete: inputs.AutocompleteInput,
+};
+
+interface IFormMetadata {
+  [value: string]: TInputMetadata;
+}
+
+export default function FormFactory(p: {
+  metadata: IFormMetadata;
+  onChange: (key: string) => (value: TInputValue) => void;
+  values: Dictionary<TInputValue>;
+  formOptions?: FormProps;
+}) {
+  const form = useForm();
+
+  useEffect(() => form.setFieldsValue(p.values), []);
```

Comment:
Not entirely sure what this is doing, but is an effect definitely what you want?  Since this is an effect, it will running after the first render and the component has been mounted in the DOM.

If you're just looking to only do something once, you use `useMemo` with an empty deps array to accomplish this synchronously.  Anyway, just wasn't sure what the semantics of this were, but in general I think we should try to reserve `useEffect` for things that need to happen after first render.

Reply:
antd was having some issues with named forms/inputs populating values on initial render for some reason, so this is basically a way to ensure that all of the proper input values are displaying correctly, it might work with useMemo, i'll give it a shot